File: GoldmanSachs/Q3.cpp
// { Driver Code Starts
#include <bits/stdc++.h>
using namespace std;



 // } Driver Code Ends


class Solution{
  public:
    int countSubArrayProductLessThanK(const vector<int>& a, int n, long long k) {
        int count = 0, start = 0, end = 1;
        long long prod = a[0];
        
        while(start<=end && end<=n){
            if(prod<k){
                count += (end-start);
                if (end<n){
                    prod*=a[end];
                }
                end+=1;
            }
            else{
                prod/= a[start];
                start+=1;
            }
        
        }
        return count;
    }
};

// { Driver Code Starts.
int main() {
    int t;
    cin >> t;
    while (t--) {
        int n, i;
        long long int k;
        cin >> n >> k;
        vector<int> arr(n);
        for (i = 0; i < n; i++) cin >> arr[i];
        Solution obj;
        cout << obj.countSubArrayProductLessThanK(arr, n, k) << endl;
    }
    return 0;
}
  // } Driver Code Ends

File: GoldmanSachs/Q3.py

class Solution:
    def countSubArrayProductLessThanK(self, a, n, k):
        #Code here
        count =0
        start = 0
        end = 1
        prod = a[0]
        while (start<=end and end<=n):
            if (prod<k):
                count += (end - start)
                if end<n:
                    prod*=a[end]
                end+=1
            else:
                prod/= a[start]
                start+=1
        return count
      
 
#{ 
#  Driver Code Starts
#Initial Template for Python 3

def main():

    T = int(input())

    while(T > 0):
        n, k = [int(x) for x in input().strip().split()]
        arr = [int(x) for x in input().strip().split()]
        
        print(Solution().countSubArrayProductLessThanK(arr, n, k))

        T -= 1


if __name__ == "__main__":
    main()


# } Driver Code Ends
